extensions/wikia/ARecoveryEngine/ARecoveryEngineApiController.class.php

```diff
@@ -13,6 +13,16 @@ public function getDelivery() {
 		$this->response->setCacheValidity( WikiaResponse::CACHE_STANDARD );
 	}
 
+	public function getPageFairDetection() {
+		$resourceLoader = new PageFairProvider();
```

Comment:
you can create a function called `getScript` and pass there either PageFairProvider or ResourceLoaderAdEngineSourcePointCSDelivery. This way you won't have to duplicate those lines:
```php
$resourceLoaderContext = new ResourceLoaderContext( new ResourceLoader(), new FauxRequest() );
$source = $resourceLoader->getScript( $resourceLoaderContext );
$this->response->setContentType( 'text/javascript; charset=utf-8' );
$this->response->setBody( $source );
$this->response->setCacheValidity( WikiaResponse::CACHE_STANDARD );